src/main/java/liadov/mypackage/lesson2/Cache.java

```diff
@@ -0,0 +1,123 @@
+package liadov.mypackage.lesson2;
+
+import java.util.Arrays;
+
+public class Cache<T> {
+    private Object[] cache;
+    private int capacity;
+    private int countElements = 0;
+
+    /**
+     * Cache constructor with initialization of cache according to received capacity
+     *
+     * @param capacity
+     */
+    public Cache(int capacity) {
+        this.capacity = capacity;
+        cache = new Object[capacity];
+    }
+
+    /**
+     * Method for adding element to Cache.
+     *
+     * @param element This element will be added to Cache
+     */
+    public void add(T element) {
+        if (countElements < capacity) {
+            cache[countElements++] = element;
+        } else {
+            cache[0] = null;
+            moveLeftAllCacheElements();
+            cache[capacity - 1] = element;
+        }
+    }
+
+    /**
+     * Method for removing element from Cache
+     *
+     * @param element This element will be removed from Cache if present
+     */
+    public void delete(T element) {
+        if (isPresent(element)) {
+            for (int i = getElementID(element); i < cache.length - 1; i++) {
+                cache[i] = cache[i + 1];
+            }
+            cache[--countElements] = null;
+        }
+    }
+
+    /**
+     * Method for checking whether element is present in Cache
+     *
+     * @param element the presence of this element is checked
+     * @return boolean
+     */
+    public boolean isPresent(T element) {
+        for (int i = 0; i < cache.length; i++) {
+            if (cache[i] != null && cache[i].equals(element)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Method returns requested element from Cache
+     *
+     * @param element this element will found in Cache and moved to top
+     * @return requested element
+     */
+    @SuppressWarnings("unchecked")
+    public T get(T element) {
+        if (isPresent(element)) {
+            for (int i = getElementID(element); i < cache.length - 1; i++) {
+                cache[i] = cache[i + 1];
+            }
+            cache[countElements - 1] = element;
+            return (T) cache[countElements - 1];
+        }
+        return null;
+    }
+
+    /**
+     * Method clean Cache
+     */
+    public void clear() {
+        for (int i = 0; i < cache.length; i++) {
+            cache[i] = null;
+        }
+        countElements = 0;
+    }
+
+    /**
+     * Method moves to the left all of Cache elements
+     */
+    private void moveLeftAllCacheElements() {
```

Comment:
У тебя ведь уже есть цикл со сдвигом, может его в метод как-то вынести и переиспользовать его вместо этого метода?

Reply:
исправляю метод так, чтобы переиспользовать:

`/**
     * Method moves to the left Cache elements
     * Elements move to the left starts from first value of idBeginMove if Method receives idBeginMove
     * else move to the left all of elements if Method receives blank idBeginMove
     * @param idBeginMove can be specified with starting index to start move. All element will be moved to the left in case of blank value
     */
    private void moveLeftCacheElements(int... idBeginMove) {
        if (idBeginMove.length == 0) {
            for (int i = 0; i < cache.length - 1; i++) {
                cache[i] = cache[i + 1];
            }
        } else {
            for (int i = idBeginMove[0]; i < cache.length - 1; i++) {
                cache[i] = cache[i + 1];
            }
        }
    }`